Add eslint as code formatter

jost-s opened the pull request:
Regarding #21 

I've added eslint and linted the code based on predominant style (single and double quotes as well as semi-colons and no semi-colons were mixed). We can swap that around depending on your preference =)

We can add prettier as well if you want.

I replaced a few unspecific types.

zippy commented on .eslintrc.js:
```diff
@@ -0,0 +1,24 @@
+  'rules': {
+    'indent': ['error', 2],
+    'linebreak-style': ['error', 'unix'],
+    'quotes': ['error', 'single'],
```
I would rather not enforce one quote style.  There are often reasons to use one vs the other, i.e. depending on if you want to quote or not quote the singe or double quotes in a particular string.

zippy reviewed (changes requested):
This is great, thank you.  Just one request, that we don't force a quote style.   I'm fine with having them cleaned up in this one PR but in general I like the flexibility so that I can do both strings like this:
```
x = "test = 'fish'"
y = '<tag attr="dog">'
```
depending on what I don't want to have to quote.

jost-s commented:
Hi @zippy, in those cases where different quotes are needed, the back tick can be used
```javascript
x = `test = 'fish'`
y = `<tag attr="dog">`
```
Nevertheless I've removed the quote rule and reverted the affected lines.

Ready for review again.

zippy reviewed (approved):
Looks great, just the packge.lock merge conflicts and ready to go.

Commit:
Add eslint & lint all js files

jost-s commented:
@zippy Super, resolved. A few more files were linted after the most recent commit.
